ledger-service/http-json/src/itlib/scala/http/AbstractWebsocketServiceIntegrationTest.scala

```diff
@@ -857,6 +857,68 @@ abstract class AbstractWebsocketServiceIntegrationTest
     case _ => -\/(jsv)
   }
 
+  "no duplicates should be returned when retrieving contracts for multiple parties" in withHttpService {
+    (uri, encoder, _) =>
+      val aliceAndBob = List("Alice", "Bob")
+      val sharedAccountNumber = "4444"
+      val jwtForAliceAndBob = jwtForParties(actAs = aliceAndBob, readAs = Nil, ledgerId = testId)
+      import spray.json._
+
+      val (killSwitch, source) = singleClientQueryStream(
+        jwt = jwtForAliceAndBob,
+        serviceUri = uri,
+        query = """{"templateIds": ["Account:SharedAccount"]}""",
+      )
+        .viaMat(KillSwitches.single)(Keep.right)
+        .preMaterialize()
+
+      def test: Sink[JsValue, Future[ShouldHaveEnded]] = {
+        val dslSyntax = Consume.syntax[JsValue]
+        import dslSyntax._
+        Consume.interpret(
+          for {
+            ContractDelta(Vector(), _, Some(liveStartOffset)) <- readOne
+            (status, _) <- liftF(
+              postCreateCommand(
+                sharedAccountCreateCommand(aliceAndBob, sharedAccountNumber),
+                encoder,
+                uri,
+                headers = headersWithPartyAuth(aliceAndBob),
+              )
+            )
+            _ = status shouldBe a[StatusCodes.Success]
+            ContractDelta(Vector((_, sharedAccountObject)), _, Some(lastSeenOffset)) <- readOne
+            _ = inside(sharedAccountObject) { case JsObject(obj) =>
+              inside((obj get "owners", obj get "number")) {
+                case (Some(JsArray(owners)), Some(JsString(number))) =>
+                  // FIXME Question for the reviewer: should we make guarantees regarding the ordering?
```

Comment:
No, it's irrelevant to the test.

Reply:
Thanks, comment removed in https://github.com/digital-asset/daml/pull/9443/commits/760b54f901b4311c4a3566ec8e3a46bf64569c6a